File: EmployeeService/Application/DTO/EmployeeDto.cs
using Domain.Enums;

namespace Application.DTO;

public class EmployeeDto : AbstractDto
{
    public int Id { get; set; }
    public string? Name { get; set; }
    public string? Surname { get; set; }
    public int Age { get; set; }
    public EmployeeGenderType Gender { get; set; }
    public DepartmentDto? Department { get; set; }
    public ProgrammingLanguageDto? ProgrammingLanguage { get; set; }
}

File: ClientApp/src/app/dto/EmployeeDto.ts
import { DepartmentDto } from "./DepartmentDto";
import { ProgrammingLanguageDto } from "./ProgrammingLanguageDto";

export interface EmployeeDto {
  id: number;
  name: string;
  surname: string;
  age: number;
  gender: number;
  department: DepartmentDto;
  programmingLanguage: ProgrammingLanguageDto;
}
